storm-core/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -0,0 +1,575 @@
+import org.apache.storm.generated.Grouping;
+import org.apache.storm.generated.SpoutSpec;
+import org.apache.storm.generated.StormTopology;
+import org.apache.storm.grouping.LoadAwareCustomStreamGrouping;
+import org.apache.storm.metric.api.IMetric;
+import org.apache.storm.metric.api.IMetricsConsumer;
+import org.apache.storm.stats.BoltExecutorStats;
+import org.apache.storm.stats.CommonStats;
+import org.apache.storm.stats.SpoutExecutorStats;
+import org.apache.storm.stats.StatsUtil;
+import org.apache.storm.task.WorkerTopologyContext;
+import org.apache.storm.tuple.AddressedTuple;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.TupleImpl;
+import org.apache.storm.tuple.Values;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.DisruptorBackpressureCallback;
+import org.apache.storm.utils.DisruptorQueue;
+import org.apache.storm.utils.Time;
+import org.apache.storm.utils.Utils;
+import org.apache.storm.utils.WorkerBackpressureThread;
+import org.json.simple.JSONValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+public abstract class Executor implements Callable, EventHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Executor.class);
+
+    protected final Map workerData;
+    protected final WorkerTopologyContext workerTopologyContext;
+    protected final List<Long> executorId;
+    protected final List<Integer> taskIds;
+    protected final String componentId;
+    protected final AtomicBoolean openOrPrepareWasCalled;
+    protected final Map stormConf;
+    protected final Map conf;
+    protected final String stormId;
+    protected final HashMap sharedExecutorData;
+    protected final AtomicBoolean stormActive;
+    protected final AtomicReference<Map<String, DebugOptions>> stormComponentDebug;
+    protected final Runnable suicideFn;
+    protected final IStormClusterState stormClusterState;
+    protected final Map<Integer, String> taskToComponent;
+    protected CommonStats stats;
+    protected final Map<Integer, Map<Integer, Map<String, IMetric>>> intervalToTaskToMetricToRegistry;
+    protected final Map<String, Map<String, LoadAwareCustomStreamGrouping>> streamToComponentToGrouper;
+    protected final ReportErrorAndDie reportErrorDie;
+    protected final Callable<Boolean> sampler;
+    protected final AtomicBoolean backpressure;
+    protected ExecutorTransfer executorTransfer;
+    protected final String type;
+    protected final AtomicBoolean throttleOn;
+    protected IFn transferFn;
+
+    protected final IReportError reportError;
+    protected final Random rand;
+    protected final DisruptorQueue transferQueue;
+    protected final DisruptorQueue receiveQueue;
+    protected Map<Integer, Task> idToTask;
+    protected final Map<String, String> credentials;
+    protected final Boolean isDebug;
+    protected final Boolean isEventLoggers;
```

Comment:
If it's same as hasEventLoggers, it would be better to keep it as it is since it represents the meaning more clearly.